topics-tools/gen-topics-database/main.go

```diff
@@ -1339,7 +1345,7 @@ func getPage(parentID int, graphVizFile io.Writer, bodyTextFile io.Writer, check
 					gvLink = strings.ReplaceAll(gvLink, "/", "_")
 					gvLinkLabel := "/" + strings.ReplaceAll(gvLink, "_", "\\n/")
 					if link.valid {
-						_, err = fmt.Fprintf(graphVizFile, "    subgraph %s {\n        %s [shape = box, color=\"#30A8A0\", style=filled, label = \"%s\"]\n    }\n", gvLink, gvLink, fmt.Sprintf("Topic - Highlighted:\\n")+gvLinkLabel)
+						_, err = fmt.Fprintf(graphVizFile, "    subgraph %s {\n        %s [shape = box, color=\"#30A8A0\", style=filled, label = \"%s\"]\n    }\n", gvLink, gvLink, fmt.Sprintf("Topic - Highlighted: %s\\n", link.linkType)+gvLinkLabel)
```

Comment:
Consider creating vars for these complex format strings.  

As someone reading this code for the first time I have no clue what this means:
`"    subgraph %s {\n        %s [shape = box, color=\"#30A8A0\", style=filled, label = \"%s\"]\n    }\n"

A well named variable will help to remove that ambiguity and keep the code short/readable.

Reply:
This code is building up a graphviz doc on the fly and putting those strings into vars would make them even harder to understand as you need to see the whole line in context ... honest - this is for what its doing time optimal.